sleap/io/format/labels_json.py

```diff
@@ -408,12 +408,35 @@ def from_json_data(
                         break
             for idx, vid in enumerate(videos):
                 for old_vid in match_to.videos:
-                    # compare last three parts of path
-                    if vid.filename == old_vid.filename or weak_filename_match(
-                        vid.filename, old_vid.filename
+
+                    # Try to match videos using either their current or source filename
+                    # if available.
+                    old_vid_paths = [old_vid.filename]
+                    if (
```

Comment:
for brevity, you could: `if getattr(old_vid.backend, "has_embedded_images", False):`